### PR TITLE
feat: import transactions from Safe builder

### PR DESCRIPTION
**Pull request by gsteenkamp89**
Fixes #UMA-2431

## motivation
We want users to be able to use the transaction builder in the safe app, export it, and then import it into the oSnap plugin.

## How to test
1. create a new Proposal (check "use oSnap" checkbox)
2. Add a new transaction
3. Choose transaction type "Safe Import" from list
4. either drag n drop or select file from machine. (see 2 test files below)
5. there are multiple transactions in the file. choose 1 to import.
6. There are 2 transaction types **Native transfer** & **contract interaction**
7. The relevant input fileds should be populated with the values from the file and validated (**it is possible to export invalid values from safe's builder**, so we must validate each input here in the app)

## Test files
This file works ✅ 
[test-file-VALID.json](https://github.com/UMAprotocol/snapshot/files/14667816/test-file-VALID.json)

This file is corrupted ❌ (should display an error message)
[test-file-INVALID.json](https://github.com/UMAprotocol/snapshot/files/14667808/test-file-INVALID.json)

screenshots:
> idle state
Screenshot 2024-03-20 at 13 43 15
> add file, choose transaction
Screenshot 2024-03-20 at 13 43 31
> Native transfer
Screenshot 2024-03-20 at 13 43 58
> Contract Interaction (includes erc20 transfers)
Screenshot 2024-03-20 at 13 43 51
> If you drag n drop a non-json file
Screenshot 2024-03-20 at 14 12 25
 > If there is an error while attempting to parse the file. can be syntax errors in the file or the schema does not match what we expect.
Screenshot 2024-03-20 at 14 13 19

### CODE REVIEW

**Comment by daywiss** [comment]
will defer reviewing this until changes from lee's suggestions

File: src/plugins/oSnap/components/Input/MethodParameter.vue [hunk 1]
```diff
@@ -67,12 +91,25 @@ function validateBytesInput(value: string) {
...
-    if (value.slice(2).length > 64) {
-      throw new Error('String too long');
+    const data = value?.slice(2) || '';
+
+    if (data.length < 64) {
+      validationErrorMessage.value = 'Value too short';
+      throw new Error('Less than 32 bytes');
+    }
+
+    if (!isBytesLike(value)) {
```
Comment (daywiss) [inline src/plugins/oSnap/components/Input/MethodParameter.vue:104]: this check should be done last.  i think it should either be in this order, too short, too long, isBytesLIke

**Review by daywiss — COMMENTED** [review]
still having an issue with deleting the first transaction in the list:
1. import valid tx file
2. remove first tx
3. see that first tx stays, but second tx is deleted

also should we disable this or let the user overwrite this tx with a new type:
image

**Comment by gsteenkamp89** [comment]
> still having an issue with deleting the first transaction in the list:
> 
> 1. import valid tx file
> 2. remove first tx
> 3. see that first tx stays, but second tx is deleted
> 
> also should we disable this or let the user overwrite this tx with a new type: image

There was some funky localised state polluting things in this flow and causing bugs like this. Now only emitting state changes upstream, much better.

File: src/plugins/oSnap/components/Input/MethodParameter.vue (hunk 1 above)
Comment (gsteenkamp89) [reply src/plugins/oSnap/components/Input/MethodParameter.vue]: fixed

File: src/plugins/oSnap/components/TransactionBuilder/TransactionImport.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+const error = ref<string>();
+
+function resetState() {
+  file.value = undefined;
+  safeFile.value = undefined;
+  error.value = undefined;
+}
+
+// TODO: allow multiple files at once
```
[the remark this answers is not in the document]
Comment (gsteenkamp89) [reply src/plugins/oSnap/components/TransactionBuilder/TransactionImport.vue]: we are only allowing them to drop 1 file **at a time**. but they can drop many, one after the other.

**Comment by daywiss** [comment]
looking good but found another issue.  after quick fixing the issues in tx 1, and then trying to simulate, it still says its inavlid:
image

**Review by daywiss — APPROVED** [review]
looks good. remember dont merge this, please pr to snapshot!

**Pull request closed without merging**